zilliqa/src/consensus.rs

```diff
@@ -1116,10 +1111,15 @@ impl Consensus {
 
         // the index is not checked here...
         // if the vote is new, store it
-        if !cosigned[index] {
-            signatures.push(new_view.signature);
-            signers.push(index as u16);
-            cosigned.set(index, true);
+        let has_voted = signatures.iter().any(|(idx, _)| *idx == index);
+
+        if !has_voted {
+            // Keeps (signer, signature) sorted by signer index in deposit contract
+            let pos = signatures
+                .binary_search_by(|(elem, _)| elem.cmp(&index))
+                .unwrap();
+            signatures.insert(pos, (index, new_view.signature));
```

Comment:
Why keep this sorted if you're going to iterate through it all with any()? (also, I'd be wary of maintaining order like this - it's far too easy for an off-by-one error - introduced by you or later by someone else - to cause the list to be "almost" sorted, and subsequent subtle "the signature was there, but then we thought it wasn't" errors).